src/openrct2/ride/RideRatings.cpp

```diff
@@ -1629,18 +1629,44 @@ static void ride_ratings_apply_max_lateral_g_penalty(
     }
 }
 
-static void ride_ratings_apply_excessive_lateral_g_penalty(rating_tuple* ratings, Ride* ride)
+static rating_tuple ride_ratings_get_excessive_lateral_g_penalty(Ride* ride)
 {
-#ifndef ORIGINAL_RATINGS
+    rating_tuple result = { 0 };
```

Comment:
This is on the stack. When returning a pointer to the tuple, it will go out of scope and may lead to corrupted memory being read at some point.